doc/ug/getting_started_build_sw.md

```diff
@@ -95,6 +95,159 @@ $ ./meson_init -t /path/to/toolchain/file
 
+## Building Software with Bazel
+
+The Mask ROM and many other software targets are built with
+(Bazel)[https://bazel.build/]
+
+TLDR:
+after installing Bazel you can run
+
+```console
+bazel test //... --test_tag_filters=-cw310 --disk_cache=~/bazel_cache
+```
+
+### Installing Bazel
+
+If you haven't yet installed Bazel, add it to apt and instal it on ubuntu systems  with the following
+commands:
+
+```console
+sudo apt install apt-transport-https curl gnupg
+curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor > bazel.gpg
+sudo mv bazel.gpg /etc/apt/trusted.gpg.d/
+echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" |
+sudo tee /etc/apt/sources.list.d/bazel.list
+sudo apt update && sudo apt install bazel-4.2.0
+```
+
+or by following [instructions for your system](https://bazel.build/install)
+though it may not be supported by OpenTitan.
+
+### Building software with Bazel
+
+You can then build and run all the tests all the Bazel rules for OpenTitan within the workspace with the command
+
+```console
+bazel test //...
+```
+
+This is likely to include unhealthy tests and targets and slow tests and targets
+so you're likely to want to run more specific builds and tests, but this
+invocation instructs bazel to build everything it needs to run all the tests in
+the workspace.
+
+Specific targets can be built with
+
+```console
+bazel build //<path to build file>:<name of target>
+```
+
+### Structure of our Bazel workspace
+
+The rules for Bazel are described in a language called starlark, which looks an
+awful lot like Python.
+
+The OpenTitan directory is defined as a bazel workspace by the
+//WORKSPACE file and BUILD files provide the information bazel needs to
+build the targets in the directories they reside in.
+
+OpenTitan uses .bzl files
+to specify rules to build targets for platforms that require specific attention
+like generation of disassembly files and different compilers.
+
+#### WORKSPACE file
+
+The Workspace file controls external dependencies such that builds can be made
+reproducible and hermetic.
+Bazel loads specific commits of external git repositories and uses them to build
+OpenTitan targets (like it does with bazel\_embedded) or to satisfy dependencies
+(as it does with abseil). As we produce increasingly stable releases we should
+fix a number of these references to specific commits and not the default branch
+head.
+
+#### BUILD files
+
+Throughout the OpenTitan repo, BUILD files describe targets and dependencies in
+the same directory and subdirectories that lack their own BUILD files. We don’t
+currently have plans to automatically generate BUILD files, and to suit the
+request we don’t include handwritten files in autogen directories, you’ll see
+some big BUILD files that describe how to build and depend on auto generated
+files in autogen directories.
+
+### Linting Starlark
+
+The Linter for Bazel files is called “buildifier” and can be built and run with
```

Comment:
```suggestion
The Linter for Bazel files is called "buildifier" and can be built and run with
```

Reply:
I'll get rid of all the smart quotes I can find

Reply:
There's a handy grep command in `ci/scripts/check-ascii.sh`. We allow non-ASCII in Markdown files at the moment, so running that script won't catch anything, but I think you probably just need something like `env LC_ALL=C xargs grep -d skip -P '[^\0-\x7f]' my-markdown-file.md`.